scripts/cbmc_json_parser.py

```diff
@@ -281,15 +286,33 @@ def modify_undefined_function_checks(properties):
     2. If a function with missing definition is reachable, then we turn all SUCCESS status to UNDETERMINED.
     If there are no reachable functions with missing definitions, then the verification is not affected, so we retain all of the SUCCESS status.
     """
-    has_unknown_location_checks = False
+    has_undefined_function_checks = False
     for property in properties:
         # Specifically trying to capture assertions that CBMC generates for functions with missing definitions
         if GlobalMessages.ASSERTION_FALSE in property["description"] and extract_property_class(
                 property) == GlobalMessages.DEFAULT_ASSERTION and not hasattr(property["sourceLocation"], "file"):
             property["description"] = "Function with missing definition is unreachable"
             if property["status"] == "FAILURE":
-                has_unknown_location_checks = True
-    return has_unknown_location_checks
+                has_undefined_function_checks = True
+    return has_undefined_function_checks
+
+def modify_expect_fail_checks(properties):
+    """
+    Invert status for Checks with the property_class "expect_fail" from FAILURE TO SUCCESS and vice versa.
+    Modify further depending on unwind/undefined function check statuses.
+    """
+    expected_fail_string = "Failed as expected."
+    unexpected_success_string = "Unexpectedly passed."
+    for property in properties:
+        if extract_property_class(property) == GlobalMessages.EXPECT_FAIL:
+            if property["status"] == GlobalMessages.STATUS_SUCCESS:
+                property["status"] = GlobalMessages.STATUS_FAILURE
+                property["description"] += ', ' + unexpected_success_string
```

Comment:
I don't think we should append it to the property description. This would make the description depend on the result which isn't the case for any other property. We should instead add it to the status.

Reply:
while we haven't done it anywhere else, would you say it's better to expand the status with an additional phrase as opposed to expanding the description? I assumed that the description would be where we could add on additional information wherever applicable.